feat: Add Revenue Tracking docs (#10952)
PostHog/posthog.com: contents/docs/web-analytics/revenue-tracking.mdx

  enterprise: full
---

import { ProductScreenshot } from 'components/ProductScreenshot'

> **Important:** This feature is currently in beta. Please leave a comment on this page if you have feedback or questions, or leave feedback via the in-app support modal.

Our revenue tracking feature enables you to capture revenue from your website via the events you are already sending us, or an integration with a payment platform, which you can then track in your [web analytics dashboard](https://app.posthog.com/web).

## Choosing a revenue data source

PostHog can collect revenue from two sources:

### Option 1: Properties on events

The simplest way to send us revenue data is to add it as properties on revenue generating events, such as when a purchase is completed, a user starts a subscription, or a subscription is renewed.

When you do so, you need to include two properties:

- `value`: The amount of revenue generated by the event.
- `currency`: The currency of the revenue in ISO 4217 format (USD, EUR, GBP, etc.).

You can choose any name you like for these properties – you can customize them in the **Revenue** tab in [data management](https://app.posthog.com/data-management/revenue).

> You `currency` property is technically optional as you can choose a static currency when configuring your revenue event in your dashboard, but we recommend sending it if you can to ensure backward compatibility should you ever take payments in a different currency.

Reviewer: The `currency` property is technically optional as you can choose a static currency when configuring your revenue event in your dashboard, but we recommend sending it if you can to ensure backward compatibility should you ever take payments in a different currency.
```

> You `currency` property is technically optional as you can choose a static currency when configuring your revenue event in your dashboard, but we recommend sending it if you can to ensure backward compatibility should you ever take payments in a different currency.

Reviewer: ```suggestion
The `currency` property is technically optional as you can choose a static currency when configuring your revenue event in your dashboard, but we recommend sending it if you can to ensure backward compatibility should you ever take payments in a different currency.
```

The `session_id` should be a UUID V7 that uniquely identifies the user's session.

You can generate a random one, but ideally you should reuse the same value we generated in our Frontend SDK, so that all events belong to the same session in our UI.

You can learn more about how to do this in the [session docs](/docs/data/sessions#server-sdks-and-sessions).

Example backend event:
```javascript
posthog.capture({
    distinct_id: posthog_distinct_id,
    event: 'Purchase Succeeded',
    properties: {
        '$session_id': posthog_session_id
    }
})
```

## Troubleshooting & FAQ

### Can I change the default reporting currency in PostHog?

Yes. You can select your desired currency in the **Revenue** tab in **Data management**. This is the currency we'll use to *report* revenue in our web analytics dashboard.

### How do you convert renvenue in different currencies?

We automatically convert revenue from other currencies into your chosen reporting currency – i.e. if someone makes a purchase in Canadian Dollars (CAD) and your reporting currency is US Dollars (USD), it's converted into USD in PostHog to ensure consistency.

> PostHog's currency conversion data is sourced from [Open Exchange Rates](https://openexchangerates.org/). The exchange rate data is updated at hourly and stored with daily granularity.

Reviewer: ```suggestion
PostHog's currency conversion data is sourced from [Open Exchange Rates](https://openexchangerates.org/). The exchange rate data is updated hourly and stored with daily granularity.
```

---
title: Revenue tracking
sidebar: Docs
showTitle: true
availability:
  free: full
  selfServe: full
  enterprise: full
---

import { ProductScreenshot } from 'components/ProductScreenshot'

> This feature is currently in beta.

Our revenue tracking feature enables you to capture revenue from your website via the events you are already sending us, or an integration with a payment platform, which you can then track in your [web analytics dashboard](https://app.posthog.com/web).

## Choosing a revenue data source

PostHog can collect revenue from two sources:

### Option 1: Properties on events

> The simplest way to send us revenue data is to add it as properties on revenue generating events, such as when a purchase is completed, a subscription is renewed, or a user subscribes to a newsletter.

Reviewer: ```suggestion
The simplest way to send us revenue data is to add it as properties on revenue generating events, such as when a purchase is completed, a subscription is renewed, or a user subscribes for the first time.
```

You can learn more about how to do this in the [session docs](/docs/data/sessions#server-sdks-and-sessions).

Example backend event:
```javascript
posthog.capture({
    distinct_id: posthog_distinct_id,
    event: 'Purchase Succeeded',
    properties: {
        '$session_id': posthog_session_id
    }
})
```

## Troubleshooting & FAQ

### Can I change the default reporting currency in PostHog?

Yes. You can select your desired currency in the **Revenue** tab in data management. This is the currency we'll **report** your revenue in our web analytics dashboard.

### How do you convert renvenue in different currencies?

We automatically convert revenue from other currencies into your chosen reporting currency – i.e. if someone makes a purchase in Canadian Dollars (CAD) and your reporting currency is US Dollars (USD), it's converted into USD in PostHog to ensure consistency.

PostHog's currency conversion data is sourced from [Open Exchange Rates](https://openexchangerates.org/). The exchange rate data is updated at hourly and stored with daily granularity.

> This ensures that your revenue data is converted using recent exchange rates, providing reliable revenue analytics across different currencies.

Reviewer 2: We should clarify that we use "the exchange rate from the time of the event" rather than "the most recent exchange rate"

> This ensures that your revenue data is converted using recent exchange rates, providing reliable revenue analytics across different currencies.

Reviewer 2: ```suggestion
This ensures that your revenue data is converted using recent exchange rates, based on the exchange rate at the time of the event, providing reliable revenue analytics across different currencies.
```

---
title: Revenue tracking
sidebar: Docs
showTitle: true
availability:
  free: full
  selfServe: full
  enterprise: full
---

import { ProductScreenshot } from 'components/ProductScreenshot'

PostHog can help you track revenue from your website via the events you are already sending us. You'll then be able to see this in your [Web Analytics dashboard](https://app.posthog.com/web).

> > This feature is currently in <span className="text-xs font-semibold text-opacity-60 bg-yellow px-1 py-0.5 rounded-sm uppercase text-primary">Beta</span>.
> 

Reviewer: ```suggestion
```

## Data source

PostHog can collect your from mainly two sources, but they're both very powerful:

### Events

> You can send us revenue tracking events from your website via the events you are already sending us.
> You can trigger an event when a purchase is completed, a subscription is renewed, or a user subscribes to a newsletter, for example.
> 
> You'll need to include two properties in this event to get the most out of it:
> - `value`: The amount of revenue generated by the event.
> - `currency`: The currency of the revenue in ISO 4217 format (USD, EUR, GBP, etc.).
> 
> You can choose any name you like for these properties, you'll be able to customize them in the [Revenue tab in Data Management](https://app.posthog.com/data-management/revenue).
> `currency` is optional as you can choose a static currency when configuring your revenue event in your dashboard.

Reviewer: ```suggestion
The simplest way to send us revenue data is to add it as properties on revenue generating events, such as when a purchase is completed, a subscription is renewed, or a user subscribes to a newsletter.

When you do so, you need to include two properties:

- `value`: The amount of revenue generated by the event.
- `currency`: The currency of the revenue in ISO 4217 format (USD, EUR, GBP, etc.).

You can choose any name you like for these properties – you can customize them in the **Revenue** tab in [data management](https://app.posthog.com/data-management/revenue).

The `currency` property is optional as you can choose a static currency when configuring your revenue event in your dashboard, but we recommend sending it if you can.
```

### Data warehouse tables

You can also expose your revenue data from your [Data Warehouse](/data-warehouse) tables.

The configuration for these is similar to the events configuration, but you'll actually need to choose 3 columns:
- `value`: The amount of revenue generated by the event.
- `currency`: The currency of the revenue in ISO 4217 format (USD, EUR, GBP, etc.).
- `timestamp`: The timestamp of when the revenue was generated.

For Data Warehouse tables, you're **required** to choose a currency from one of its columns. This restriction might be lifted in the future. We don't support custom SQL at the moment.

We currently have a built-in connection to Stripe, which will allow you to import your Stripe payment data directly into PostHog.

> Integrations with other payment platforms such as RevenueCat and LemonSqueezy are coming soon.

Author: We could, we aren't really sure what we're building next, this is mostly a hidden way to figure out what is it that customers want, I'm happy to add it.

> Integrations with other payment platforms such as RevenueCat and LemonSqueezy are coming soon.

Reviewer: We can leave it for now, I think. Stripe will cover most users and if people ask for Chargebee instructions, we can add them later.

> 
> #### Integrating with Stripe

Reviewer: ```suggestion
```

You'll first want to connect your Stripe account to PostHog. You can follow our [Stripe integration docs](/tutorials/stripe-reports) to do so.

Once you've connected your Stripe account, you'll be able to choose the `stripe_price` table and the `amount`, `currency` and `created_at` columns from the Stripe data.

> And that's all, you'll immediately see your revenue data in your Web Analytics dashboard.

Author: That's all automatic when you connect to Stripe, all of these tables and columns show up inside PostHog.

> And that's all, you'll immediately see your revenue data in your Web Analytics dashboard.

Reviewer: Does this mean that, once you connect Stripe, revenue starts appearing in the web analytics dashboard automatically without any further configuration?

The instructions imply there's some config, but also say it happens immediately. It's just a bit unclear to me atm.

> And that's all, you'll immediately see your revenue data in your Web Analytics dashboard.

Author: Ah, I understand it now. The flow is the following:

- They connect with Stripe following the tutorials (linked in there)
- Then inside the same tab as the one from events (/data-warehouse/revenue) they can select to connect with a data warehouse table (`stripe_price`) and point to the correct columns (`amount`, `currency` and `created_at`)

How would you phrase it?

> And that's all, you'll immediately see your revenue data in your Web Analytics dashboard.

Author: thank you :)
